strongbox-commons/src/main/java/org/carlspring/strongbox/artifact/coordinates/AbstractArtifactCoordinates.java

```diff
@@ -72,7 +73,7 @@ protected final String setCoordinate(String coordinate,
 
     public Map<String, String> getCoordinates()
     {
-        return Collections.unmodifiableMap(coordinates);
+        return new HashMap<>(coordinates);
```

Comment:
I'm not a very big fan of  `HashMap` and `HashSet`, as you have no guarantee regarding the order of the elements in them. Is there a particular reason to be using this instead of `LinkedHashMap`? If there is no particular reason, then would you mind replacing this here and below (I've seen a few other places in the changed code that uses `HashMap` and `HashSet`). If I'm missing the reason, could you please elaborate.

Reply:
I agree, but also I am not sure this will be safe and not break a lot of UnitTests.
maybe we can assign this for OSS guys?

Reply:
Let's not leave it for the OSS guys. I think it should be safe to just change the ones from your pull.

Reply:
ok
done

Reply:
Also I think we should go away from current approach which is to store coordinates as Map in OrientDB. We should use just private fields for them instead.